automata/pda/pda.py

```diff
@@ -2,17 +2,40 @@
 """Classes and methods for working with all pushdown automata."""
 
 import abc
-from typing import AbstractSet, Literal
+import os
+import random
+import uuid
+from collections import defaultdict
+from typing import (
+    AbstractSet,
+    Any,
+    DefaultDict,
+    Generator,
+    List,
+    Literal,
+    Optional,
+    Tuple,
+    Union,
+)
 
 import automata.base.exceptions as exceptions
 import automata.pda.exceptions as pda_exceptions
 from automata.base.automaton import Automaton, AutomatonStateT, AutomatonTransitionsT
+from automata.base.utils import LayoutMethod, create_graph, save_graph
 from automata.pda.configuration import PDAConfiguration
 from automata.pda.stack import PDAStack
 
+# Optional imports for use with visual functionality
+try:
+    import coloraide
+    import pygraphviz as pgv
+except ImportError:
+    pass
```

Comment:
Should use the same error message as in other files if the imports are not present.